### ISSUE TITLE
 OpenTok:Publisher:error onPublishingTimeout

### ISSUE THREAD

**Issue by Andrei1982** [opentok/opentok-react#173]
Hi, today I'm faced with a row of errors when the client tries to connect the establishment:

`OpenTok:Publisher:error OT.Publisher.onPublishingTimeout +0ms` 
```
OpenTok:GlobalExceptionHandler:error OT.exception :: title: Unable to Publish (1500) msg: ICEWorkflow
index.js:3535
``` 
`OpenTok:Session:error 1500 +0ms Session.publish :: Could not publish in a reasonable amount of time`

Reproduced for me on the Chrome v. 88.0.4324.96 (Official Build) (x86_64) / MacOS Mojave 10.14.5

On the Chrome Canary or FF not reproduced.
Also not reproduced on the Chrome v. 88.0.4324.96 (Official Build) (x86_64) / Windows 10

Any thoughts? Need help!

**Comment by kamenivskyi** [comment]
I have the same problem on Iphone Safari v.14.

**Comment by kamenivskyi** [comment]
@Andrei1982 have you resolved your error?

**Comment by enricop89** [comment]
@kamenivskyi are you able to reproduce it consistently? It sounds like a network error to me. Can you try with a different network, please? You can also inspect the session using https://tokbox.com/developer/tools/inspector/

**Comment by enricop89** [comment]
Yes, in iOS Safari you can't have two active getUserMedia at the same time. In your case, you have `<OTPublisher>` and the `detectDevices` function above. What's the goal of function anyway? For the device selection you can use `navigator.mediaDevices.enumerateDevices()`

**Comment by kamenivskyi** [comment]
@enricop89 well, thank you very much! :). I just tried to do like in this example: https://opentok.github.io/opentok-web-samples/Publish-Devices/js/app.js and get devices like this: 
```
 await OT.getDevices((err, devices) => {
      if (err) {
          console.log('error occurred in getAvailableDevices func: , ', err);
          reject(err);
      }
      const 
audioDevices = devices.filter((device) => device.kind === 'audioInput');
      const videoDevices = devices.filter((device) => device.kind === 'videoInput');

      if (videoDevices && audioDevices) {
          resolve({ video: videoDevices, audio: audioDevices });
      }
  });
 ``` 
inside the OT.getUserMedia() promise.
 I will try with navigator.mediaDevices.enumerateDevices

**Comment by enricop89** [comment]
OT.getDevices is a wrapper of `navigator.mediaDevices.enumerateDevices`

**Comment by ilyadesyatov** [comment]
I have same problem with chromebook
